src/WebJobs.Extensions.ApiHub/Config/ApiHubConfiguration.cs

```diff
@@ -83,10 +88,18 @@ private IFolderItem GetFileSource(string key)
         /// Add path to the configuration
         /// </summary>
         /// <param name="key">App settings key name that have the connections string</param>
-        /// <param name="connectionString">Connection string to access SAAS via ApiHub. <seealso cref="ApiHubJobHostConfigurationExtensions.GetApiHubProviderConnectionStringAsync"/></param>
-        public void AddKeyPath(string key, string connectionString)
+        /// <param name="connectionString">Connection string to access SAAS via ApiHub. <seealso cref="ApiHubJobHostConfigurationExtensions.GetApiHubProviderConnectionStringAsync" /></param>
+        /// <param name="writer">The writer.</param>
+        public void AddKeyPath(string key, string connectionString, TraceWriter writer = null)
         {
-            var root = ItemFactory.Parse(connectionString);
+            ApiHubLogger logger = null;
+
+            if (writer != null)
+            {
+                logger = new ApiHubLogger(writer);
```

Comment:
I don't think you want to add the TraceWriter param to the AddKeyPath method. Users aren't going to know how to new one of these up and pass it in. Anyhow, if a parse exception is thrown, it will already be logged to the log files. I just verified this myself by throwing an exception in ApiHubConfiguration.ApplyToConfig